accepted/00083-containerized-proxy.md

```diff
@@ -0,0 +1,82 @@
+- Feature Name: Deliver Proxy as a container
+- Start Date: 2021-10-11
+
+# Summary
+[summary]: #summary
+
+Deliver Proxy as a container application.
+
+# Motivation
+[motivation]: #motivation
+
+There are deployment scenarios in which adding a new Virtual Machine to implement Proxy functionality is cumbersome or impossible, while leveraging an existing container runtime cold be an easier option, especially in highly distributed ("Edge") environments.
+
+Having the option of deploying Proxy functionality as a container can be easier to operate (eg. deployment, independence from underlying OS, keeping up to date) and more flexible (eg. implementing active/passive HA by leveraging an existing Kubernetes environment).
+
+It is believed implementing Proxy containerization will surface a subset of common problems and solutions also useful to implement a containerized Server in future.
+
+# Detailed design
+[design]: #detailed-design
+
+## Delivery
+  - Proxy is delivered as one container
+    - Updates are delivered via new images on a public Registry
+  - Documentation is produced to describe how this container can be run on popular conainer runtimes: [docker](https://docs.docker.com/engine/reference/commandline/cli/) and [podman](https://podman.io/)
+  - A minimal [Helm](https://helm.sh/) chart is delivered and tested against [k3s](https://k3s.io/)
+
+## Onboarding
+  - Containerized Proxy is not a managed client, does not run `salt-minion`, is not registered in the usual way
+  - A new UI page (and API endpoint) allows to add containerized Proxies by specifying their FQDN
+    - After confirmation a System entry (in the System list) is created 
+      - Normal client actions (package management, configuration file management...) will not be available on that System
+      - Proxy-specific Server UI/API functionality, like showing the list of attached clients, will be available like on regular Proxies
+    - for each such System, the Server generates a tarball ("configuration directory") that can be downloaded
+    - Proxy container needs the configuration directory to be mounted as a volume. After start, it is fully connected and operational
+
+## Operating
+  - Container will expose Salt (4505/4506) and https (443) ports
```

Comment:
I guess specifying ports will be part of the config in tarball, but IMO we should default for containers to listen on some higher ports and then do mapping in docker/podman/some ingress service.

I think it is reasonable to assume that there will be multiple services/containers running on one machine and they may want to use https as well.

Reply:
Yes, sorry. I always mix up expose and port publishing.